src/binary.ts

```diff
@@ -277,24 +193,18 @@ export class Binary {
 
   /** @internal */
   toJSON(): string {
-    if (!this.buffer) return '';
-    const buffer = Buffer.from(this.buffer as Uint8Array);
-    return buffer.toString('base64');
+    return this.buffer.toString('base64');
   }
 
   /** @internal */
   toString(format: BufferEncoding): string {
-    if (!this.buffer) return '';
-    const buffer = Buffer.from(this.buffer.slice(0, this.position) as Uint8Array);
-    return buffer.toString(format);
+    return this.buffer.slice(0, this.position).toString(format);
```

Comment:
I know you didn't write this, but it's not clear why we want to slice the buffer. I think you can just have `return this.buffer.toString(format)`

Reply:
I think it is because `this.position` represents the end of written data, without it the toString will return the whole buffer, which can be larger than the data we are concerned about.